app/Console/Commands/FlightawareImport.php

```diff
@@ -0,0 +1,249 @@
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class FlightawareImport extends Command {
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'Realops:LoadFromFlightaware';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Flightaware import task. Interactive and VERY SLOW.';
+
+    public function calculateChunks($start_time, $chunk_count, $optimal_chunk_len_seconds) {
+        $pbar_chunks = $this->output->createProgressBar($chunk_count * 2);
+        $pbar_chunks->start();
+
+        $chunks_unix = [];
+        $chunks = [];
+
+        // This is, admittedly, really janky.
+        // Doing it with Carbon didn't work though.
+        // This does, so ¯\_(ツ)_/
+
+        $current_time = $start_time->timestamp;
+
+        for ($i = 0; $i < $chunk_count; $i++) {
+            $old_time = $current_time;
+            $current_time += $optimal_chunk_len_seconds;
+            $new_time = $current_time;
+
+            $chunks_unix[] = [$old_time, $new_time];
+
+            $pbar_chunks->advance();
+        }
+
+        for ($y = 0; $y < $chunk_count; $y++) {
+            $chunks[] = [Carbon::createFromTimestamp($chunks_unix[$y][0]), Carbon::createFromTimestamp($chunks_unix[$y][1])];
+            $pbar_chunks->advance();
+        }
+
+        $pbar_chunks->finish();
+        $this->info(''); // blank line after progress bar
+
+        return $chunks;
+    }
+
+    public function save_flights($pbar, $flights) {
+        for ($i = 0; $i < sizeof($flights); $i++) {
+            $flight = $flights[$i];
+            $pbar->setMessage('Saving ' . $flight['ident']);
+
+            $callsign = $flight['ident'];
+
+            $existing = RealopsFlight::where('flight_number', $callsign)->first();
+            if ($existing !== null) {
+                continue;
+            }
+            $new_flight = new RealopsFlight;
+
+            // parse the departure and arrival times
+            $deptime = $flight['scheduled_out'];
+            $arrtime = $flight['scheduled_in'];
+            $deptime_parsed = Carbon::parse($deptime);
+            $arrtime_parsed = Carbon::parse($arrtime);
+
+            $new_flight->assigned_pilot_id = null;
+            $new_flight->flight_number = $callsign;
+            $new_flight->flight_date = $deptime_parsed->toDateString();
+            $new_flight->dep_time = $deptime_parsed->toTimeString();
+            $new_flight->dep_airport = $flight['origin'];
+            $new_flight->arr_airport = $flight['destination'];
+            $new_flight->est_arr_time = $arrtime_parsed->toTimeString();
+            $new_flight->route = 'Pilot Choice';
+
+            $new_flight->save();
+            $pbar->advance();
+        }
+    }
+
+    public function load_departures(&$flights, $ratelimit_max_flights, $ratelimit_timeout, $client, $chunk, $pbar, &$flights_count) {
```

Comment:
Never seen it in PHP before, but `&$flights` and `&$flights_count` passes by reference like in C? So that doesn't happen automatically for non-primitive objects like Java? Or do arrays counts as primitives in PHP?

Reply:
Spot on! It drastically simplifies these functions so I chose to use it here. [Manpage](https://www.php.net/manual/en/language.references.pass.php)

Reply:
According to the manual, everything is passed as a copy by default, but this is PHP after all so I figure there are some exceptions to this.

Reply:
Facts. Works the same as C - you have to prefix with & to pass by reference